### PR TITLE
Add missing functional tests for CIS AWS 1.16, 1.17, 1.20

### PR DESCRIPTION
**Pull request by orestisfl**
### Related Issues

- https://github.com/elastic/csp-security-policies/issues/223
- https://github.com/elastic/csp-security-policies/issues/220
- https://github.com/elastic/csp-security-policies/issues/222

### Checklist
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added the necessary README/documentation (if appropriate)

### CODE REVIEW

File: tests/product/tests/data/aws/aws_iam_test_cases.py [hunk 1]
```diff
@@ -230,6 +233,35 @@
     "1.15 Permissions through groups: inline policy, expect: failed": cis_aws_iam_1_15_fail,
 }
 
+cis_aws_iam_1_16 = {
+    "1.16 Built-in EC2 policy, expect: passed": EksAwsServiceCase(
+        rule_tag=CIS_1_16,
+        case_identifier="AmazonEC2FullAccess",
+        expected=RULE_PASS_STATUS,
+    ),
```
Comment (gurevichdmitry) [inline tests/product/tests/data/aws/aws_iam_test_cases.py:241]: @orestisfl, this test case (1.16) fails -> expected passed retrieved failed. Please check again if expected result is correct.

File: tests/product/tests/data/aws/aws_iam_test_cases.py [hunk 2]
```diff
@@ -230,6 +233,35 @@
...
+}
+
+cis_aws_iam_1_20 = {
+    "1.20 Valid access analyzers in all regions, expect: passed": EksAwsServiceCase(
+        rule_tag=CIS_1_20,
+        case_identifier="account-access-analyzers",
+        expected=RULE_PASS_STATUS,
+    ),
+}
+
```
Comment (gurevichdmitry) [inline tests/product/tests/data/aws/aws_iam_test_cases.py:264]: @orestisfl, this test case (1.20) fails -> no evaluation found even after 10 minutes. Please check why it happens.

**Review by gurevichdmitry — CHANGES_REQUESTED** [review]
@orestisfl, great. Please see my comments.

File: tests/product/tests/data/aws/aws_iam_test_cases.py (hunk 1 above)
Comment (orestisfl) [reply tests/product/tests/data/aws/aws_iam_test_cases.py:241]: Should be fixed by https://github.com/elastic/csp-security-policies/pull/228

File: tests/product/tests/data/aws/aws_iam_test_cases.py (hunk 2 above)
Comment (orestisfl) [reply tests/product/tests/data/aws/aws_iam_test_cases.py:264]: Should be fixed by https://github.com/elastic/cloudbeat/pull/916 & https://github.com/elastic/csp-security-policies/pull/230

File: tests/product/tests/data/aws/aws_iam_test_cases.py (hunk 1 above)
Comment (orestisfl) [reply tests/product/tests/data/aws/aws_iam_test_cases.py:241]: @gurevichdmitry please check again

File: tests/product/tests/data/aws/aws_iam_test_cases.py (hunk 2 above)
Comment (orestisfl) [reply tests/product/tests/data/aws/aws_iam_test_cases.py:264]: @gurevichdmitry please check again

[the remark this answers is not in the document]
Comment (orestisfl) [reply tests/product/tests/data/aws/aws_iam_test_cases.py:264]: 1.17: Ensure a support role has been created to manage incidents with AWS Support
1.20: Ensure that IAM Access analyzer is enabled for all regions

Given that these rules are once-per-account, I don't see a way to both have them fail & succeed. Any ideas?

[the remark this answers is not in the document]
Comment (gurevichdmitry) [reply tests/product/tests/data/aws/aws_iam_test_cases.py:264]: It appears that we need an additional account. I created a common task (https://github.com/elastic/security-team/issues/6204) to cover all problematic test cases that require the use of an additional account. Could you please update this task?

**Review by gurevichdmitry — APPROVED** [review]
approved

File: tests/product/tests/data/aws/aws_iam_test_cases.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (orestisfl) [reply tests/product/tests/data/aws/aws_iam_test_cases.py:264]: Done, thanks!

**Pull request merged**